data.py

```diff
@@ -35,119 +36,185 @@ def set_context(resource_item):
+        validate_and_parse_json(g, schemas.Editions, edition, "editions")
 
     # ------------------
     # Versions resources
     # ------------------
 
-    versions_source_path = Path(
-        subbed_metadata_store_content_path / "editions/versions"
-    )
+    # Load json files from disk
+    versions = process_json_files(versions_source_path)
 
-    # Validate then add to graph
-    schema = schemas.Versions
-    process_json_files(g, versions_source_path, schema)
+    # Validate data and add to graph
+    for version in versions:
+        assert (
+            version["@id"] in versions_in_edition.keys()
+        ), f"Versions URL {version['@id']} not found in {versions_in_edition.keys()}"
+        summarised_versions = [
+            {"@id": vsn["@id"], "issued": vsn["issued"]} for vsn in version["versions"]
+        ]
+        for summarised_version in summarised_versions:
+            assert (
+                summarised_version in versions_in_edition[version["@id"]]
+            ), f"Discrepancy between {versions_in_edition} and {summarised_versions}"
+        validate_and_parse_json(g, schemas.Versions, version, "versions")
 
     # ------------------
     # Topics resources
     # ------------------
 
-    topics_source_path = Path(subbed_metadata_store_content_path / "topics.json")
+    # Load json file from disk
     with open(topics_source_path) as f:
         topics_source_dict = json.load(f)
-        # Validate then add to graph
-        schemas.Topics(**topics_source_dict)
-        g += Graph().parse(
-            data=json.dumps(set_context(topics_source_dict)),
-            format="json-ld",
-        )
-    assert len(g) > graph_length
-    graph_length = len(g)
+
+    # Validate data and add to graph
+    topic_ids = [topic["@id"] for topic in topics_source_dict["topics"]]
+    for dataset in datasets_source_dict["datasets"]:
+        for topic in dataset["topics"]:
+            assert (
+                topic in topic_ids
+            ), f"{topic} not in list of approved topics {topic_ids}"
+    for topic in topics_in_editions:
+        assert topic in topic_ids, f"{topic} not in {topic_ids}"
+    validate_and_parse_json(g, schemas.Topics, topics_source_dict, "topics")
 
     # --------------------
     # Publishers resources
     # --------------------
 
-    publishers_source_path = Path(
-        subbed_metadata_store_content_path / "publishers.json"
-    )
+    graph_length = len(g)
     with open(publishers_source_path) as f:
         publishers_source_dict = json.load(f)
-        # Validate then add to graph
-        schemas.Publishers(**publishers_source_dict)
-        g += Graph().parse(
-            data=json.dumps(set_context(publishers_source_dict)),
-            format="json-ld",
-        )
+
+    # TODO This currently fails as OFCOM is not listed as a publisher in any of the datasets (it's a creator for 4gc). Do we need a separate creators.json?
```

Comment:
This failure definitely seems beyond the scope/unrelated to this task so I would probably keep this comment here and consider a new ticket for it.